build/references/command-line-interface.md

```diff
@@ -254,65 +254,168 @@ Enables peer specific query latency metrics. Defaults to `false`.
 
 Minimum amount of time messages to a peer must be failing before the peer is benched. Defaults to `5m`.
 
-### C-Chain / Coreth
+### Chain Configs
 
-`--coreth-config` \(json\):
+AvalancheGo can read chain-specific configs from external files. VM interface have 2 fields named `configBytes` and `upgradeBytes`. These are supplied with contents in the `config` and `upgrade` files.
 
-This allows you to specify a config to be passed into the C-Chain. The default values for this config are:
+AvalancheGo looks for sub-folders named with chain IDs or chain aliases, under chain config directory. Each sub-folder may include two files named `config` or `upgrade`. Their extensions do not matter as their contents will be handled by their own VMs. However if more than one file is provided with the same name, i.e both `config.json` & `config.txt` in the same sub-folder, AvalancheGo will give error.
+
+AvalancheGo always precedes chain ID over chain alias, similarly primary alias precedes over other aliases. All folder & file names are case sensitive.
+
+`--chain-config-dir` \(string\):
+
+Specifies root folder for chain configs, it defaults to `$HOME/.avalanchego/configs/chains/`. If the flag is not set and this default directory is not available (not exist), it won’t give any; error since configs are optional. However if the flag is set, the pointed folder must exists, otherwise it gives error.
```

Comment:
this default directory does not exist, it will not exit with an error since configs are optional.

Reply:
if the flag is set, the specified folder must exist, or AvalancheGo will exit with an error.